cpp/src/io/orc/aggregate_orc_metadata.cpp

```diff
@@ -172,9 +172,10 @@ std::vector<metadata::stripe_source_mapping> aggregate_orc_metadata::select_stri
       // Coalesce stripe info at the source file later since that makes downstream processing much
       // easier in impl::read
       for (const auto& stripe_idx : user_specified_stripes[src_file_idx]) {
-        CUDF_EXPECTS(stripe_idx < static_cast<decltype(stripe_idx)>(
-                                    per_file_metadata[src_file_idx].ff.stripes.size()),
-                     "Invalid stripe index");
+        CUDF_EXPECTS(
+          stripe_idx >= 0 and stripe_idx < static_cast<decltype(stripe_idx)>(
+                                             per_file_metadata[src_file_idx].ff.stripes.size()),
+          "Invalid stripe index");
```

Comment:
I suggest that we could upcast the `auto=cudf::size_type` to match the `size_t` that the `size()` function is returning, rather than play type gymnastics by downcasting the result of `size()` (requiring a `decltype()`) and also checking `>= 0`? The end result appears to be shorter and safer. (Excuse the lack of clang-format)
```suggestion
        CUDF_EXPECTS(static_cast<std::size_t>(stripe_idx) < per_file_metadata[src_file_idx].ff.stripes.size(), "Invalid stripe index");
```

Reply:
but then we would be static casting to size_t a value that can be negative. IMO this option works "by accident", the explicit check here is preferable to me.

Reply:
Can a stripe index be negative? A `size_type==int` can be negative but that doesn't mean that `stripe_idx` should ever take on negative values (unless I'm unaware of how it's used). In the same way, we would need to know in the current snippet that the `size()` call returning a `size_t` won't overflow a `size_type` when being cast. I would argue that the purported safety of _either option_ is conditional on prior knowledge of the values, and is not an intrinsic guarantee of the choice of type/casting.

Reply:
Stripe index is a value passed by the user, so we need to check if it's in valid range. It is imposible for size to be more than max size_type, since it's always (way) smaller than the number of rows.

Reply:
> Stripe index is a value passed by the user

Okay, that's what I was missing. I assumed it was a loop index or something where we had stronger guarantees about its potential values.

Reply:
Yeah, this check is just input validation, which is what the failing test is validating.